src/context.cpp

```diff
@@ -454,24 +455,48 @@ namespace librealsense
         auto callback_id = unique_id::generate_id();
         _devices_changed_callbacks.insert(std::make_pair(callback_id, std::move(callback)));
 
+        if (_device_watcher_active == false)
```

Comment:
I think this flag isn't needed.
You should be able to ask the device watcher if it's started or not and act accordingly.

Reply:
done